include/ginkgo/core/preconditioner/batch_jacobi.hpp

```diff
@@ -0,0 +1,221 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_PRECONDITIONER_BATCH_JACOBI_HPP_
+#define GKO_PUBLIC_CORE_PRECONDITIONER_BATCH_JACOBI_HPP_
+
+
+#include <ginkgo/core/base/array.hpp>
+#include <ginkgo/core/base/batch_lin_op.hpp>
+#include <ginkgo/core/base/batch_multi_vector.hpp>
+#include <ginkgo/core/base/exception_helpers.hpp>
+#include <ginkgo/core/base/lin_op.hpp>
+#include <ginkgo/core/base/types.hpp>
+#include <ginkgo/core/matrix/batch_csr.hpp>
+#include <ginkgo/core/matrix/batch_dense.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+
+
+namespace gko {
+namespace batch {
+namespace preconditioner {
+
+
+/**
+ * A block-Jacobi preconditioner is a block-diagonal linear operator, obtained
+ * by inverting the diagonal blocks (stored in a dense row major fashion) of the
+ * source operator.
+ *
+ * With the batched preconditioners, it is required that all items in the batch
+ * have the same sparsity pattern. The detection of the blocks and the block
+ * pointers require that the sparsity pattern of all the items be the same.
+ * Other cases is undefined behaviour. The input batch matrix must be in
+ * batch::Csr matrix format or must be convertible to batch::Csr matrix format.
+ * The block detection algorithm and the conversion to dense blocks kernels
+ * require this assumption.
+ *
+ * @note In a fashion similar to the non-batched Jacobi preconditioner, the
+ * maximum possible size of the diagonal blocks is equal to the maximum warp
+ * size on the device (32 for NVIDIA GPUs, 64 for AMD GPUs).
+ *
+ * @tparam ValueType  value precision of matrix elements
+ * @tparam IndexType  index precision of matrix elements
+ *
+ * @ingroup jacobi
+ * @ingroup precond
+ * @ingroup BatchLinOp
+ */
+template <typename ValueType = default_precision, typename IndexType = int32>
+class Jacobi final : public EnableBatchLinOp<Jacobi<ValueType, IndexType>> {
+    friend class EnableBatchLinOp<Jacobi>;
+    friend class EnablePolymorphicObject<Jacobi, BatchLinOp>;
+
+public:
+    using EnableBatchLinOp<Jacobi>::convert_to;
+    using EnableBatchLinOp<Jacobi>::move_to;
+    using value_type = ValueType;
+    using index_type = IndexType;
+    using matrix_type = batch::matrix::Csr<ValueType, IndexType>;
+
+    /**
+     *  Returns the block pointers.
+     *  @note Returns nullptr in case of a scalar jacobi preconditioner
+     *        (max_block_size = 1).
+     *  @return the block pointers
+     */
+    const index_type* get_const_block_pointers() const noexcept
+    {
+        return block_pointers_.get_const_data();
+    }
+
+    /**
+     * Returns the mapping between the blocks and the row id.
+     *
+     * @note Returns nullptr in case of a scalar jacobi preconditioner
+     *       (max_block_size = 1).
+     */
+    const index_type* get_const_row_block_map_info() const noexcept
```

Comment:
now that the doc has been cleared up, I guess it could be renamed to `get_const_map_block_to_row` (or other way around) (then the member could also be renamed)